doc.go

```diff
@@ -21,6 +21,27 @@
 //	// Run tests...
 //
 //	err = framework.DeleteCluster(ctx, cluster)
+
+// # Example using an existing Workload Cluster
+//
+//	ctx := context.Background()
+//
+//	framework, err := clustertest.New("context_name")
+//	if err != nil {
+//		panic(err)
+//	}
+//
+//	// The E2E_WC_NAME and E2E_WC_NAMESPACE env vars must be exported
+//	cluster, err := framework.LoadCluster()
+//	if err != nil {
+//		panic(err)
+//	}
+//	if cluster == nil {
+//		// Handle the case where the env vars aren't provided
+//	}
```

Comment:
Do you think a `LoadOrCreateCluster` helper function that wraps around this whole thing would be useful? 
E.g. rather than every test suite needing to add this block they instead change the current `ApplyCluster` call with this new wrapper that will use the existing if found.

Reply:
I thought about it but couldn't come up with a nice interface for it. You would need to pass either the whole cluster you want to create or just the path to the manifests and a provider, but also get back the cluster that was **actually** created. It was a bit confusing.
Do you have something else in mind? I'm open to ideas, but I think we should probably do it in a separate PR

Reply:
I agree that it can come in another PR so lets not block this one. 

I don't have any solid idea. I think lets try using this and see what feels right.